crypto/s2n_ecc_evp.c

```diff
@@ -437,11 +434,11 @@ int s2n_ecc_evp_write_params_point(struct s2n_ecc_evp_params *ecc_evp_params, st
 
     POSIX_GUARD(s2n_ecc_evp_calculate_point_length(point, group, &point_len));
     S2N_ERROR_IF(point_len != ecc_evp_params->negotiated_curve->share_size, S2N_ERR_ECDHE_SERIALIZING);
-    point_blob.data = s2n_stuffer_raw_write(out, point_len);
-    POSIX_ENSURE_REF(point_blob.data);
-    point_blob.size = point_len;
 
+    POSIX_GUARD(s2n_alloc(&point_blob, point_len));
     POSIX_GUARD(s2n_ecc_evp_write_point_data_snug(point, group, &point_blob));
+    POSIX_GUARD(s2n_stuffer_write(out, &point_blob));
+    POSIX_GUARD(s2n_free(&point_blob));
```

Comment:
Allocating and then immediately freeing is pretty inefficient. Can you figure out a way to do this without an allocation?